--- vmcs_sm.c	2017-09-13 00:17:57.000000000 +0200
+++ vmcs_sm_fixed.c	2017-09-14 12:56:48.085094712 +0200
@@ -197,13 +197,15 @@
 static struct SM_STATE_T *sm_state;
 static int sm_inited;
 
+#if __GNUC__ < 6 
+
 static const char *const sm_cache_map_vector[] = {
 	"(null)",
 	"host",
 	"videocore",
 	"host+videocore",
 };
-
+#endif
 /* ---- Private Function Prototypes -------------------------------------- */
 
 /* ---- Private Functions ------------------------------------------------ */
